fvm/evm/handler/addressAllocator.go

```diff
@@ -73,13 +51,20 @@ func (aa *AddressAllocator) AllocatePrecompileAddress(index uint64) types.Addres
 }
 
 func MakePrecompileAddress(index uint64) types.Address {
-	return makePrefixedAddress(index, FlowEVMPrecompileAddressPrefix)
+	return makePrefixedAddress(index, types.FlowEVMExtendedPrecompileAddressPrefix)
 }
 
-func makePrefixedAddress(index uint64, prefix [addressPrefixLen]byte) types.Address {
+func makePrefixedAddress(
+	index uint64,
+	prefix [types.FlowEVMSpecialAddressPrefixLen]byte,
+) types.Address {
 	var addr types.Address
 	prefixIndex := types.AddressLength - uint64ByteSize
 	copy(addr[:prefixIndex], prefix[:])
 	binary.BigEndian.PutUint64(addr[prefixIndex:], index)
 	return addr
 }
+
+func shuffleAddressIndex(preShuffleIndex uint64) uint64 {
+	return uint64(preShuffleIndex * addressIndexShuffleSeed)
+}
```

Comment:
@tarakby just checking if this is okay?

Reply:
Thanks for the tag, I'll have a look at this shortly

Reply:
I made suggestions in this PR https://github.com/onflow/flow-go/pull/5344